File: app/src/main/java/com/frankegan/verdant/fullscreenimage/FullscreenImagePresenter.java
package com.frankegan.verdant.fullscreenimage;

import com.frankegan.verdant.models.ImgurImage;

/**
 * Created by frankegan on 5/18/16.
 */
public class FullscreenImagePresenter implements FullscreenImageContract.UserActionsListener {

    ImgurImage imageModel;
    FullscreenImageContract.View fullscreenView;

    public FullscreenImagePresenter(ImgurImage imageModel, FullscreenImageContract.View fullscreenView) {
        this.imageModel = imageModel;
        this.fullscreenView = fullscreenView;
        if (imageModel.getAnimated())
            fullscreenView.setGif(imageModel.getThumbLink());
        else
            fullscreenView.setImage(imageModel.getThumbLink());
    }

    @Override
    public void lightsOut() {
        // TODO: 5/18/16 look up how to go lights out
    }
}

File: app/src/main/java/com/frankegan/verdant/fullscreenimage/FullscreenImagePresenter.kt
package com.frankegan.verdant.fullscreenimage

import com.frankegan.verdant.models.ImgurImage

/**
 * Created by frankegan on 5/18/16.
 */
class FullscreenImagePresenter(internal var imageModel: ImgurImage,
                               internal var fullscreenView: FullscreenImageContract.View) :
        FullscreenImageContract.UserActionsListener {

    init {
        if (imageModel.animated) {
            fullscreenView.setGif(imageModel.thumbLink)
        } else {
            fullscreenView.setImage(imageModel.thumbLink)
        }
    }

    override fun lightsOut() {
        // TODO: 5/18/16 look up how to go lights out
    }
}
